### PR TITLE
add variable action

### PR DESCRIPTION
**Pull request by dawehner**
not done yet

### CODE REVIEW

**Comment by fago** [comment]
please also create a d.o. issue and reference it at https://www.drupal.org/node/2269577

File: src/Plugin/Action/AddVariable.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * )
+ */
+class AddVariable extends RulesActionBase {
+
+  /**
+   * Executes the plugin.
+   */
+  public function execute() {
+    $this->setProvidedValue('variable_added', $this->getContext('value'));
+    return TRUE;
```
Comment (klausi) [inline src/Plugin/Action/AddVariable.php:35]: This method should not return anything.

File: src/Tests/AddVariableIntegrationTest.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+namespace Drupal\rules\Tests;
+
+/**
+ * Test the data processor plugins during Rules evaluation.
+ *
+ * @group rules
+ */
+class AddVariableIntegrationTest extends RulesDrupalTestBase {
```
Comment (klausi) [inline src/Tests/AddVariableIntegrationTest.php:15]: Why do we need a Simpletest for this and cannot use the PHPUnit test base? Please add a comment.

**Comment by dasjo** [comment]
this is the according issue https://www.drupal.org/node/2409049

**Comment by dasjo** [comment]
see the follow-up https://github.com/fago/rules/pull/121

**Pull request closed without merging**